src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -462,9 +462,11 @@ protected function hasSelectStatement(QueryBuilder $queryBuilder, $statement)
      */
     protected function assignGroupBy($queryBuilder)
     {
-        if (!empty($this->groupByFields)) {
-            foreach ($this->groupByFields as $fields) {
-                $queryBuilder->groupBy($fields->getSelect());
+        $groupByFields = array_merge($this->groupByFields, $this->sortFields);
```

Comment:
At current state we have 3 queries:

1. id query
2. select query
3. group query

Think from performance point of view, I think the only one which requires to sort by `id` is the `id query`. And the others one don't require sorting and or filters as they just require `WHERE id IN (...)`. And the rest can be handled on the PHP side.

Reply:
because i am quite sure that we will introduce another issue with that 😂  and i am also not really sure if sorting in php is faster that doing it in the queries :)

What i have done is removing the sort by from the grouping query. It is really unnecessary because we getting the results index by the id so the sort itself is made for nothing